includes/plugins/woocommerce/my-account/class-my-account-ui-v1.php

```diff
@@ -119,5 +130,481 @@ public static function my_account_menu_items( $items ) {
 		unset( $items['customer-logout'] );
 		return $items;
 	}
+
+	/**
+	 * Remove required fields from the account settings form.
+	 *
+	 * @param array $required_fields The required fields.
+	 * @return array The required fields.
+	 */
+	public static function account_settings_required_fields( $required_fields ) {
+		unset( $required_fields['account_display_name'] );
+		return $required_fields;
+	}
+
+	/**
+	 * Intercept a password reset request from Account Settings page.
+	 * The key must be generated before the user's password can be reset,
+	 * but not before submitting the form, otherwise it will break the Forgot Password flow.
+	 */
+	public static function maybe_generate_password_reset_key() {
+		// Only if the user is logged in and a reader.
+		$user = \wp_get_current_user();
+		if ( ! \is_user_logged_in() || ! Reader_Activation::is_user_reader( $user ) ) {
+			return;
+		}
+
+		// Only if updating password from Account Settings page.
+		$action = filter_input( INPUT_POST, 'action', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		$nonce  = filter_input( INPUT_POST, 'woocommerce-reset-password-nonce', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		if ( self::RESET_PASSWORD_ACTION !== $action || empty( $nonce ) ) {
+			return;
+		}
```

Comment:
Shouldn't we also verify the nonce before moving forward?

Reply:
It's already being checked by Woo, but we probably shouldn't assume. Updated in 459eb1c2